### PR TITLE
Fix resource code generation

### PR DESCRIPTION
**Pull request by dsyme**
Fixes https://github.com/Microsoft/visualfsharp/issues/3689

@brettfo You had to check for an empty string, not just a null.  

This bug was found by running some of the extra "Mono" tests we've historically run in http://github.com/fsharp/fsharp. Note we should activate these tests in this repository, particularly the ones building all the projects under ``tests\projects``

### CODE REVIEW

**Comment by dsyme** [comment]
The fix is this line her https://github.com/Microsoft/visualfsharp/pull/3690/files#diff-4cd33802b25af2df1dea8575f4be65bcR135

File: src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs [hunk 1]
```diff
@@ -134,7 +132,7 @@ module internal {1} =
                     if getBooleanMetadata "GenerateSource" false item then
                         let moduleName =
                             match item.GetMetadata("GeneratedModuleName") with
-                            | null -> Path.GetFileNameWithoutExtension(item.ItemSpec)
+                            | null | "" -> Path.GetFileNameWithoutExtension(item.ItemSpec)
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs]: perhaps use string.IsNullOrWhitspace and if then else. the match looks a bit hokey here.

**Comment by dsyme** [comment]
Merging as this is blocking for updates to fsharp/fsharp and FCS

File: src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs (hunk 1 above)
Comment (dsyme) [reply src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs]: yes would be better

**Pull request merged**